src/java.base/share/classes/java/lang/Math.java

```diff
@@ -3421,8 +3421,8 @@ public static float scalb(float f, int scaleFactor) {
     }
 
     // Constants used in scalb
-    static double twoToTheDoubleScaleUp = powerOfTwoD(512);
-    static double twoToTheDoubleScaleDown = powerOfTwoD(-512);
+    private static final double twoToTheDoubleScaleUp = powerOfTwoD(512);
+    private static final double twoToTheDoubleScaleDown = powerOfTwoD(-512);
```

Comment:
Aren't these the _literals_ `0x1p512` and `0x1p-512`, respectively?

Reply:
Whoa. You are right.
```
    public static void main(String[] args) {
        System.out.println(twoToTheDoubleScaleUp);
        System.out.println(twoToTheDoubleScaleDown);
        System.out.println(0x1p512);
        System.out.println(0x1p-512);

        System.out.println(twoToTheDoubleScaleUp == 0x1p512);
        System.out.println(twoToTheDoubleScaleDown == 0x1p-512);
    }
```

```
1.3407807929942597E154
7.458340731200207E-155
1.3407807929942597E154
7.458340731200207E-155
true
true
```

Reply:
These static fields seem to be used just once each, so we might as well just replace the usages with the literals and get rid of the fields.